index.js

```diff
@@ -32,3 +32,19 @@ module.exports = fn => {
 		};
 	};
 };
+
+module.exports.proxy = function (fn) {
+	return id => {
+		const target = function () {};
+		const handler = {
+			get: function (target, property) {
+				return fn(id)[property];
```

Comment:
You need to cache the function call so it's only called once, like in the existing implementation.